src/progress-indicator/progress-indicator.interface.ts

```diff
@@ -0,0 +1,35 @@
+import { TemplateRef } from "@angular/core";
+
+export interface Step {
+	/**
+	 * The label of the `step`
+	 */
+	text: string;
+	/**
+	 * An array of strings to determine the progress of the step
+	 */
+	state: Array<string>;
+	/**
+	 * Defines the tooltip
+	 */
+	tooltip?: Tooltip;
+	/**
+	 * Determines whether the step is disabled or not
+	 */
+	disabled?: boolean;
+}
+
+export interface Tooltip {
```

Comment:
This may collide with the actual `Tooltip` export ... I'd go with something like `ProgressIndicatorTooltip` ... or maybe something less verbose 😬